src/Extension/Extension.php

```diff
@@ -96,8 +96,9 @@ public function __construct($path, $composerJson)
      */
     protected function assignId()
     {
-        $segments = explode('/', $this->path);
-        $this->id = end($segments);
+        list($vendor, $package) = explode('/', $this->name);
+        $package = str_replace(['flarum-ext-', 'flarum-'], null, $package);
```

Comment:
Hmm, minor thing: It surprised me that the second parameter to `str_replace` can be `null`. Can you change it to an empty string? That's really what we want to express here, right?